File: output/java/HexToAnsi.java
public class HexToAnsi {
    public static String hexToAnsi(String hex) {
        hex = hex.replace("#", "");
        int r = Integer.parseInt(hex.substring(0, 2), 16);
        int g = Integer.parseInt(hex.substring(2, 4), 16);
        int b = Integer.parseInt(hex.substring(4, 6), 16);
        return String.format("\033[38;2;%d;%d;%dm", r, g, b);
    }

    public static void main(String[] args) {
        System.out.println(hexToAnsi("#FF4500") + "Orange text" + "\033[0m");
    }
}

File: output/rust/hex_to_ansi.rs
fn hex_to_ansi(hex: &str) -> String {
    let hex = hex.trim_start_matches('#');
    let r = u8::from_str_radix(&hex[0..2], 16).unwrap();
    let g = u8::from_str_radix(&hex[2..4], 16).unwrap();
    let b = u8::from_str_radix(&hex[4..6], 16).unwrap();
    format!("\x1b[38;2;{};{};{}m", r, g, b)
}

fn main() {
    println!("{}Orange text[0m", hex_to_ansi("#FF4500"));
}
